### ISSUE TITLE
In argument detected as Inout

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#198]
When a function uses a temporary variable whose size is determined by an argument then any modification of this variable leads to the size being marked as an inout variable.

The following is a minimum working example:
```
from pyccel.decorators import types, external

@external
@types('int')
def test_degree(degree):
    from numpy import empty

    tmp = empty(degree)
    for i in range(5):
         tmp[i]=0.
```

It generates the following fortran:
```
subroutine test_degree(degree)

  implicit none
  integer(kind=4), intent(inout)  :: degree
  real(kind=8), allocatable :: tmp (:)
  integer(kind=4) :: i




  allocate(tmp(0:degree - 1))
  do i = 0, 4, 1
    tmp(i) = 0.0d0
  end do

end subroutine
```

Possibly related to issue #174

**Comment by yguclu** [comment]
As @ratnania explained in a recent meeting, this is happening because `degree` is an argument to Numpy's function `empty`. Currently we assume that any argument to a Numpy factory function (like `empty`, `zeros` and `ones`) has intent `inout`, and this information is propagated back to the user function signature. Hence Pyccel assumes that `empty` must be an `inout` argument of `test_degree` in this case.

This problem can be solved by keeping a database with the intent of the various arguments to the Numpy functions supported by Pyccel. @saidctb: We think you can easily take care of this, do you have time? As usual we need a separate branch, a few unit tests and a pull request.

**Comment by saidctb** [comment]
yes I will take care of it

**Comment by saidctb** [comment]
the bug was in the `get_assigned_symbols` it returns the wrong results  because we used `free_variables` of sympy we should use `atoms` instead in general.

**Comment by yguclu** [comment]
Fixed by pull request #214
